From 71d15ad7a1da9da8d0d85cf1a128c8c01e7c74c8 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Thu, 14 Jan 2021 10:16:15 -0800
Subject: [PATCH 1/5] util: Fix pointer to integer conversion error when using
 libunwind

../src/util/u_debug_stack.c:97:20: error: incompatible pointer to integer conversion assigning to 'unw_word_t' (aka 'unsigned long') from 'void *' [-Werror,-Wint-conversion]
   pip.unwind_info = NULL;
                   ^ ~~~~
1 error generated.

Fixes: 70c272004f72 "gallium/util: libunwind support"
Closes: https://gitlab.freedesktop.org/mesa/mesa/-/issues/4094

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/8507>
(cherry picked from commit 08ac7b52e7814540e92d5add29049324f6841bb5)
---
 src/util/u_debug_stack.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/util/u_debug_stack.c b/src/util/u_debug_stack.c
index 21f0371d3c1..01f69e14442 100644
--- a/src/util/u_debug_stack.c
+++ b/src/util/u_debug_stack.c
@@ -94,7 +94,7 @@ debug_backtrace_capture(struct debug_stack_frame *backtrace,
    unw_proc_info_t pip;
    unsigned i = 0;
 
-   pip.unwind_info = NULL;
+   pip.unwind_info = 0;
 
    unw_getcontext(&context);
    unw_init_local(&cursor, &context);
-- 
2.29.2 (Apple Git-129)

